stack-graphs/src/stitching.rs

```diff
@@ -754,18 +829,18 @@ impl<H: Clone> ForwardPartialPathStitcher<H> {
     /// Attempts to extend one partial path as part of the algorithm.  When calling this function,
     /// you are responsible for ensuring that `db` already contains all of the possible appendables
     /// that we might want to extend `partial_path` with.
-    fn extend<A, Db>(
+    fn extend<A, Db, C, Err>(
         &mut self,
-        graph: &StackGraph,
-        partials: &mut PartialPaths,
-        db: &mut Db,
+        candidates: &mut C,
```

Comment:
Only API suggestion I can think of is to consider if this could/should be `mut candidates: C` instead.  I _think_ you're always passing in a `mut` ref to a temporary, in which case you can just pass in that temporary directly.  It probably doesn't matter from a codegen perspective, though, so if you consider this shape to be more understandable I'm also good with keeping it as-is.

Reply:
I tried this, but the `SQLiteReader` implements `Candidates` directly, and would be moved (and lost) by this change. So I'm going to leave it as-is.